adder/single/dag_service.go

```diff
@@ -62,19 +68,14 @@ func (dgs *DAGService) Add(ctx context.Context, node ipld.Node) error {
 func (dgs *DAGService) Finalize(ctx context.Context, root cid.Cid) (cid.Cid, error) {
 	// Cluster pin the result
 	rootPin := api.PinWithOpts(root, dgs.pinOpts)
-	rootPin.Allocations = dgs.dests
+	if dgs.local {
+		return root, adder.Pin(ctx, dgs.rpcClient, rootPin)
+	}
```

Comment:
The allocations should be respected. We are block-putting locally, but the file should be pinned normally as it would have been pinned if we were block-putting to other places.

Reply:
pin operation would take care of that since we care passing pin options with rootPin,
Checked it to be respecting options.

Reply:
https://github.com/ipfs/ipfs-cluster/pull/907/commits/ca81048ffa0b3cd62dc5e44c158f640f26264eec